// @cognigy/extension-tools/build/createToolDescriptor.spec.js
"use strict";
/**
 * This is not an ordinary (unit-)test file, but a file that just
 * verifies that we can properly transpile the code and use a tool
 * descriptor.
 */
Object.defineProperty(exports, "__esModule", { value: true });
const createToolDescriptor_1 = require("./createToolDescriptor");
// Successfully creates a tool descriptor
(0, createToolDescriptor_1.createToolDescriptor)({
    type: "httpRequest",
    defaultLabel: "HTTP Request",
    defaultDescription: "Sends an HTTP request to an external system.",
    defaultParameters: [
        {
            key: "method",
            label: "HTTP Method",
            description: "Which HTTP method to use. Valid values are GET, POST, PUT, DELETE, and PATCH",
            defaultValue: "GET",
            type: "select",
            params: {
                required: true,
                options: [
                    { label: "GET", value: "GET" },
                    { label: "POST", value: "POST" },
                    { label: "PUT", value: "PUT" },
                    { label: "DELETE", value: "DELETE" },
                    { label: "PATCH", value: "PATCH" },
                ]
            }
        },
        {
            key: "urL",
            label: "Request URL",
            description: "The URL to send the HTTP request to.",
            defaultValue: "",
            type: "text",
            params: {
                required: true,
            }
        },
        {
            key: "headers",
            label: "HTTP Headers",
            description: "Optional HTTP headers to include in the request.",
            defaultValue: {},
            type: "keyValuePair"
        },
        {
            key: "payload",
            label: "Request Payload",
            description: "The payload to include in the request body (for POST, PUT, and PATCH requests).",
            defaultValue: "{}",
            type: "json"
        }
    ],
    function: async ({ parameters }) => {
        const { method, url, headers, payload } = parameters;
        const response = await fetch(url, {
            method,
            headers,
            body: ["POST", "PUT", "PATCH"].includes(method) ? payload : undefined
        });
        const data = await response.json();
        return {
            status: response.status,
            statusText: response.statusText,
            data
        };
    }
});
